### PR TITLE
Situation search 2

### PR DESCRIPTION
**Pull request by weradsaoud**

### CODE REVIEW

File: src/standalone/popups/offender-criteria-popup/offender-criteria-popup.component.html [hunk 1]
```diff
@@ -250,28 +250,22 @@ <h4 class="mb-4 text-primary">{{ lang.map.search_criteria }}</h4>
...
-                        })
-                      "
-                      icon="INFORMATION"></app-icon-button>
-                  </div>
+                  <app-situation-search-btn
+                    [id]="
+                      element.type === offenderTypesEnum.EMPLOYEE
+                        ? element.offenderInfo?.id ?? element.id
+                        : element.offenderInfo?.agentId ?? element.agentId
+                    "
```
Comment (aalshahin) [inline src/standalone/popups/offender-criteria-popup/offender-criteria-popup.component.html:258]: [id]="
                      element.type === offenderTypesEnum.EMPLOYEE
                        ? element.id
                        : element.agentId
                    "
is working fine.

File: src/standalone/popups/offender-criteria-popup/offender-criteria-popup.component.html [hunk 2]
```diff
@@ -250,28 +250,22 @@ <h4 class="mb-4 text-primary">{{ lang.map.search_criteria }}</h4>
...
-                      (click)="
-                        situationSearch$.next({
-                          offender: element,
-                          isCompany: true
-                        })
-                      "
-                      icon="CLEARING_AGENCY"></app-icon-button>
-                  </div>
+                  <app-situation-search-btn
+                    [id]="element.offenderInfo?.agencyId ?? element.agencyId"
```
Comment (aalshahin) [inline src/standalone/popups/offender-criteria-popup/offender-criteria-popup.component.html:265]: [id]="element.agencyId"
is working fine.

File: src/standalone/components/offender-list/offender-list.component.html [hunk 3]
```diff
@@ -77,13 +77,15 @@ <h1 class="text-primary text-xl">{{ lang.map.offenders }}</h1>
...
-                  situationSearch$.next({ offender: element, isCompany: false })
-                "
-                icon="INFORMATION"></app-icon-button>
-            </div>
+            <app-situation-search-btn
+              [id]="
+                element.type === offenderTypes.EMPLOYEE
+                  ? element.offenderInfo?.id ?? element.id
+                  : element.offenderInfo?.agentId ?? element.agentId
+              "
```
Comment (aalshahin) [inline src/standalone/components/offender-list/offender-list.component.html:85]: [id]="
                element.type === offenderTypes.EMPLOYEE
                  ? element.offenderInfo?.id
                  : element.offenderInfo?.agentId
              "
              
              will work fine

File: src/standalone/components/offenders-violations-preview/offenders-violations-preview.component.html [hunk 4]
```diff
@@ -93,10 +93,15 @@
...
-          <app-icon-button
-            [matTooltip]="lang.map.situation_search"
-            (click)="situationClick($event, element, false)"
-            icon="INFORMATION"></app-icon-button>
+          <app-situation-search-btn
+            [id]="
+              element.type === offenderTypes.EMPLOYEE
+                ? element.offenderInfo?.id ?? element.id
+                : element.offenderInfo?.agentId ?? element.agentId
+            "
```
Comment (aalshahin) [inline src/standalone/components/offenders-violations-preview/offenders-violations-preview.component.html:101]: same as offender list component

**Review by aalshahin — CHANGES_REQUESTED** [review]

File: src/modules/electronic-services/components/situation-search-btn/situation-search-btn.component.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  dialog = inject(DialogService);
+
+  @Input({ required: true })
+  isCompany!: boolean;
+  @Input({ required: true })
+  id!: number;
+  @Input({ required: true })
+  type!: number;
+  @Input({ required: true })
+  icon: keyof AppIconsType = 'RELOAD';
```
[the remark this answers is not in the document]
Comment (weradsaoud) [reply src/modules/electronic-services/components/situation-search-btn/situation-search-btn.component.ts:37]: ok, I made it INFORMATION

File: src/standalone/components/offender-list/offender-list.component.html (hunk 3 above)
Comment (weradsaoud) [reply src/standalone/components/offender-list/offender-list.component.html:85]: done

File: src/standalone/popups/offender-criteria-popup/offender-criteria-popup.component.html (hunk 1 above)
Comment (weradsaoud) [reply src/standalone/popups/offender-criteria-popup/offender-criteria-popup.component.html:258]: done

**Pull request closed without merging**